### DISCUSSION TITLE
Steam Deck implementation bug (I almost got it working)

### DISCUSSION THREAD

**Discussion by lyndonguitar** [LizardByte/community#96] [Sunshine]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your issue described in the documentation?

- [X] I have read the documentation

### Describe the Bug

I just got my steam deck a few days ago, and did a bunch of stuff including trying out Sunshine.

Since there is a flatpak available for Sunshine, I tried that, however it is not available on Discover so you have to install manually.

It worked in desktop mode. I can use Sunshine and broadcast from Steam Deck as server, to clients (Tested a Tablet and Phone with Moonlight streaming client)

However, the challenge lies in making it work for Game Mode. Since I have previous experience in forcing an app to run in game mode at startup (the app is syncthing, which usually only works in desktop mode too) I figured I'll try this with Sunshine too.

Using the same technique, I created a service on systemd folder that runs Sunshine, and started that service (it means on every boot it will run, just like a windows service).

I'm about halfway successful. while in game-mode, Sunshine is indeed running in the background. My clients can detect it. However this is where the problem lies, as soon as I connect, the connection is instantly terminated. So it definitely has something to do with how gamemode UI handles it compared to desktop.

Also, when going back to desktop mode after doing this. looks like Sunshine messes up Steam Deck Desktop's resolutions settings, as it now displays a blackscreen, persistent on restarts.

### Expected Behavior

Connection is instantly terminated when connecting... it should connect successfully.

It also messes up Steam Deck Desktop's resolutions settings, as it now displays a blackscreen, persistent on restarts.

### Additional Context

Sunshine/Moonlight performs way better than stock Steam Remote Play protocol.

### Operating System Version

SteamOS 3.3.2

### Architecture

64 bit

### Sunshine commit or version

v0.15.0

### Package

Linux - flatpak

### GPU Type

AMD

### GPU Model

AMD Custom GPU 0405 (Vangogh)

### GPU Driver/Mesa Version

4.6 (Compatibility Profile) Mesa 22.0.2 (git-71648faf15)

**Comment by istori1** [comment]
Try running it as a system service and add this to the service.
[Unit]
StartLimitBurst=4

[Service]
Environment="WAYLAND_DISPLAY=gamescope-0"
ExecStart=flatpak run dev.lizardbyte.sunshine
ExecStop=flatpak kill dev.lizardbyte.sunshine
Restart=on-failure
RestartSec=1s

**Comment by lyndonguitar** [comment]
> Try running it as a system service and add this to the service. [Unit] StartLimitBurst=4
> 
> [Service] Environment="WAYLAND_DISPLAY=gamescope-0" ExecStart=flatpak run dev.lizardbyte.sunshine ExecStop=flatpak kill dev.lizardbyte.sunshine Restart=on-failure RestartSec=1s

Thanks.

The commands doesn't seem to run sunshine either in desktop or game mode.

Removing "Environment="WAYLAND_DISPLAY=gamescope-0" makes it run it again, but with the problems above.

EDIT:
Switching to AppImage version had better results. The connection is finally not terminated immediately during game mode, BUT I can only see black screen and a mouse cursor at the center. (Desktop mode still works without problems, as usual)

It still doesnt run sunshine when "Environment="WAYLAND_DISPLAY=gamescope-0" is included in the commands. 

EDIT2: Additional things I've tried that didnt work:
setting an advertised resolution of 1280x800 and 60fps
setting an application shortcut that is blank
setting an application shortcut with the ff. Command Preparations
xrandr --output HDMI-1 --mode 1280x800
selecting to other encoders (cannot detect encoder in using other than software)

**Comment by ReenigneArcher** [comment]
Instead of duplicating efforts and conversations, I'd recommend you start here.

- https://github.com/LizardByte/Sunshine/discussions/370 OR
- https://discord.com/channels/804382334370578482/1029527413333430313/1029527413333430313

Also, I'm moving this to a discussion instead of an issue since Sunshine doesn't currently support game mode on steam deck.

**Comment by AlvaroRojas** [comment]
I'm also interested in making this work as steam link latency is pretty high.
